### PR TITLE
Add Python 3.9 to GHA

### PR DESCRIPTION
**Pull request by gliptak**
Signed-off-by: Gábor Lipták <gliptak@gmail.com>

### CODE REVIEW

File: .github/workflows/test-python-package.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.6, 3.7, 3.8]
+        python-version: [3.6, 3.7, 3.8, 3.9, 3.10]
```
Comment (JGSweets) [inline .github/workflows/test-python-package.yml:16]: I believe `3.10` requires  quotes around it: https://github.com/actions/setup-python/issues/160

**Comment by JGSweets** [comment]
@gliptak Seems like py3.10 has an issue installing python-snappy?

**Comment by gliptak** [comment]
https://github.com/andrix/python-snappy/issues/113

**Comment by JGSweets** [comment]
I think the error lies in the test:
`test_cannot_instantiate` within `dataprofiler.tests.profilers.test_column_profile_compilers.TestBaseProfileCompilerClass`

Need to catch the case where method becomes methods between versions.

**Review by JGSweets — APPROVED** [review]

**Review by taylorfturner — APPROVED** [review]
LGTM

**Pull request merged**